nft/nft_lib/src/main.sw

```diff
@@ -0,0 +1,139 @@
+library nft_lib;
+
+use std::{
+    address::Address,
+    assert::require,
+    constants::NATIVE_ASSET_ID,
+    chain::auth::{AuthError, Sender, msg_sender},
+    result::*,
+    revert::revert,
+};
+
+enum Error {
+    AddressCannotBeZero: (),
+    NFTNotInitalized: (),
+    OwnerAlreadySet: (),
+    OwnerCannotBeApproved: (),
+    SenderNotOwner: (), 
+}
+
+// TODO: Add mappings when supported for token ids
+
+struct Data {
+    owner: Address,
+    approved: Address,
+}
+
+storage {
+    data: Data,
+    state: u64,
```

Comment:
Can we make this an enum?  It would be more readable.

Reply:
I don't believe that an enum makes sense here

Reply:
My thinking is that an enum may be preferable to track the state of the contract instead of using integers.  So instead of state being 0 to represent the uninitialized state it would be like State::Uninitialized.  Or something like that.

Reply:
enums are not supported in storage the last time I have checked so this is a workaround for now